--- a/deps/zlib/zlib.gyp
+++ b/deps/zlib/zlib.gyp
@@ -190,6 +190,12 @@
             'include_dirs': [ '<(ZLIB_ROOT)' ],
           },
           'conditions': [
+            ['OS=="android"', {
+              # cpu_features.c calls the NDK cpufeatures API directly.
+              'sources': [
+                '<(ZLIB_ROOT)/android_cpu_features.c',
+              ],
+            }],
             ['OS!="win"', {
               'cflags!': [ '-ansi' ],
               'cflags': [ '-Wno-implicit-fallthrough' ],
